library/src/test/java/com/datatorrent/lib/filter/FilterTest.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.datatorrent.lib.filter;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Tests for FilterOperator
+ */
+public class FilterTest
+{
+  private static final Logger logger = LoggerFactory.getLogger(FilterTest.class);
+
+  public static class DummyPrivatePOJO
+  {
+    private long val;
+
+    public long getVal()
+    {
+      return val;
+    }
+
+    public void setVal(long val)
+    {
+      this.val = val;
+    }
+  }
+
+  public static class DummyPublicPOJO
+  {
+    public long val;
+  }
+
+
+  private static FilterOperator filter;
+  private static DummyPrivatePOJO data;
+  private static DummyPublicPOJO pdata;
+
+  @Test
+  public void testFilterPrivate()
+  {
+    filter.inClazz = DummyPrivatePOJO.class;
+    filter.setCondition("({$}.getVal() == 100)");
```

Comment:
This test is specifying Private member variable in condition. {$}.val raises following exception: Private member cannot be accessed from type "SC".